bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyColorUtils.java

```diff
@@ -172,8 +172,9 @@ public static PercentType toPercent(Integer value) {
         return toPercent(value, 0, SHELLY_MAX_COLOR);
     }
 
-    public static PercentType toPercent(@Nullable Integer _value, Integer min, Integer max) {
+    public static PercentType toPercent(Integer _value, Integer min, Integer max) {
         Double range = max.doubleValue() - min.doubleValue();
+        @SuppressWarnings("null")
         Double value = _value != null ? _value.doubleValue() : 0;
```

Comment:
Shouldn't the null checking been handled already by the static analyzer. Why do you need to check again? Or better question: why did you make the parameter non-null?

Reply:
> That was pretty harsh to go through. I wish you split this up into smaller PRs.
> I'd rather not review this entire thing again.

I'm very sorry, The PR grow over time, because refactoring was a bigger effort. Along the line I added features. I already thought about how to manage that better in the future. The problem is: Due to a lack of a complete set of devices I need to test with other people, which could take a longer time to get feedback. During that time I continue to work on the binding...

Thanks for taking this challenge.